File: 84512. 모음 사전/모음 사전.java
import java.util.HashMap;

class Solution {
    final String[] alphabets = {"A", "E", "I", "O", "U"};
	HashMap<String, Integer> wordDic = new HashMap<>();
    
    public void setDic(String str){
		int length = str.length();
		wordDic.put(str, wordDic.size());
		
		if(length > 4) return;
		for(int i = 0; i < alphabets.length; i++) {
			setDic(str + alphabets[i]);
		}
	}
    public int solution(String word) {
		setDic("");
		
		int answer = wordDic.get(word);
        return answer;
    }
}

File: 1주차/84512. 모음 사전/모음 사전.py
alphabets = ['A', 'E', 'I', 'O', 'U']
wordDic = {}
def setDic(str):
    wordDic[str] = len(wordDic)
    if len(str) > 4: return
    
    for i in range(0, 5):
        setDic(str + alphabets[i])

def solution(word):
    answer = 0
    setDic("")
    
    answer = wordDic[word]
    return answer